### PR TITLE
new adaptations for gizmos

### PR DESCRIPTION
**Pull request by YunHsiao**
added run-time VB update support, among other adaptations.

### CODE REVIEW

File: cocos/3d/assets/texture-cube.js [hunk 1]
```diff
@@ -23,138 +23,58 @@
...
-  height: 2,
-  format: gfx.TEXTURE_FMT_RGBA8,
-  anisotropy: 1,
-  wrapS: gfx.WRAP_REPEAT,
-  wrapT: gfx.WRAP_REPEAT,
-  minFilter: gfx.FILTER_LINEAR,
-  magFilter: gfx.FILTER_LINEAR,
-  mipFilter: gfx.FILTER_LINEAR,
+let isComplete = function(elements) {
+    return elements.reduce((acc, t) => t = t || acc.complete);
```
Comment (dabingnn) [inline cocos/3d/assets/texture-cube.js]: I love this iterative implementation. but it seems isComplete returns true only if all elements is complete. should we use `t && acc.complete`?

File: cocos/3d/framework/model-component.js [hunk 2]
```diff
@@ -87,7 +87,6 @@ let ModelShadowCastingMode = Enum({
  */
 @ccclass('cc.ModelComponent')
 @executionOrder(100)
-@executeInEditMode
```
Comment (dabingnn) [inline cocos/3d/framework/model-component.js:90]: It seems if we do not use `@executeInEditMode`, then `onEnable` `onDisable` will not be called in editor.  model will show wrongly. Am I right @jareguo ?

File: cocos/3d/builtin/init.js [hunk 3]
```diff
@@ -194,24 +167,23 @@ export default function (device) {
...
-    // ].forEach(name => {
-    //     let mat = new Material();
-    //     mat.effectAsset = effects[`builtin-effect-${name}`];
-    //     mat._uuid = `builtin-material-${name}`;
-    //     mat._loaded = true;
-    //     materials[mat._uuid] = mat;
-    // });
-
-    //
+    [
```
Comment (dabingnn) [inline cocos/3d/builtin/init.js]: given we do not use it, suggest we comment it out now.

File: cocos/3d/assets/texture-cube.js (hunk 1 above)
Comment (YunHsiao) [reply cocos/3d/assets/texture-cube.js]: oops :P

**Review by dabingnn — APPROVED** [review]

**Pull request merged**